src/Stratis.Bitcoin/FinalizedBlockInfoRepository.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using DBreeze;
+using DBreeze.DataTypes;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin
+{
+    /// <summary>Provider of the last finalized block's height and hash.</summary>
+    /// <remarks>
+    /// Finalized block height is the height of the last block that can't be reorged.
+    /// Blocks with height greater than finalized height can be reorged.
+    /// <para>Finalized block height value is always <c>0</c> for blockchains without max reorg property.</para>
+    /// </remarks>
+    public interface IFinalizedBlockInfoRepository : IDisposable
+    {
+        /// <summary>Gets the finalized block hash and height.</summary>
+        /// <returns>Hash and height of a block that can't be reorged away from.</returns>
+        HashHeightPair GetFinalizedBlockInfo();
```

Comment:
This could return null wich might crash the caller.

Reply:
No.
Only if we didn't call `LoadFinalizedBlockInfoAsync` which is a part of initialization.